common/types.go

```diff
@@ -278,6 +280,47 @@ func (l *LifecycleEvent) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// MarshalText returns the string representation of the enum value.
+// It returns an error if the enum value is invalid.
+func (l *LifecycleEvent) MarshalText() ([]byte, error) {
+	if l == nil {
+		return []byte(""), nil
+	}
+	var (
+		ok bool
+		s  string
+	)
+	if s, ok = lifecycleEventToString[*l]; !ok {
+		return nil, fmt.Errorf("invalid lifecycle event: %v", int(*l))
+	}
+
+	return []byte(s), nil
+}
+
+// UnmarshalText unmarshals a text representation to the enum value.
+// It returns an error if given a wrong value.
+func (l *LifecycleEvent) UnmarshalText(text []byte) error {
```

Comment:
> I thought it made more sense implemented in UnmarshalText()

Care to explain the why to your clueless coworker? :)

Reply:
Because we need to do this conversion anyway, and validating the value in `UnmarshalText()` makes more sense to me than having a standalone validation function, which will also probably return the converted value if it passes.

But like I said, since we'll want to move this over to `enumer`, it might be better to validate it separately, but then again, it's not a particularly big deal to leave it for later. :man_shrugging:

Reply:
Thanks for explaining. It makes sense to do validation near the type itself.

I usually use `UnmarshalText` if another part of the code expects a text/json unmarshaller. Otherwise, it becomes noisy fast as follows:

```
lifeCycle := opts.Get(k).String()
if err := o.WaitUntil.UnmarshalText([]byte(lifeCycle)); err != nil {
	return fmt.Errorf("error parsing goto options: %w", err)
}
```

Reply:
Besides the `[]byte()` conversion, I don't see how that's noisy. :man_shrugging: In this case I'd rather follow an established interface than have a custom validation function. But we're discussing style and personal preference now, which we agreed not to do. :)

We'll think about this once we start using `enumer`, so I'll merge this as is.